### DISCUSSION TITLE
chap 10 질문 정리

### DISCUSSION THREAD

**Discussion by inseo24** [JS-OS-study/hyokyung-and-students#34] [General]
파일 시스템

**Comment by inseo24** [comment]
<h3>느낀 점 & 궁금한 점 </h3>

- 파일 시스템 생각보다 딥해서 놀랬다. 다양하게 많은 것을 배운 거 같다.
- Page Cache and Buffer Cache 쪽은 집중력이 떨어져서 나중에 다시 봐야할 거 같다.

<hr />
<h3>문제</h3>

Q. 유닉스 파일 시스템은 디스크 파일 할당 방식에서 어떤 것이 기초가 되었을까요?(영문 한글 무관)

**Reply by 1chz** [reply]
Linked Allocation

**Reply by sihyung92** [reply]
Linked Allocation -> 파일의 메타데이터를 담고 있는 Inode라는 독립공간 생성

**Reply by inseo24** [reply]
indexed allocation

**Comment by 1chz** [comment]
# 느낀점

---

- 장소만 메모리에서 디스크로 바뀌었다 뿐이지 메모리 관리와 생각보다 큰 차이가 없다고 느꼈다
- 자원 관리를 하드웨어들이 하는 것과 운영체제가 개입 되는 것의 차이가 크다고 느꼈다

<br />

# 문제

---

- 메모리 관리와 다르게 파일 시스템에서 LRU, LFU를 사용할 수 있는 이유는 무엇인지?

**Reply by inseo24** [reply]
메모리는 하드웨어에서의 작용이 중요한데 파일 시스템은 system call로 운영체제가 관여할 수 있어서?

**Reply by sihyung92** [reply]
항상 open() 시 시스템콜이 일어나므로, 운영체제에서 (가장 최근 / 가장 빈번)에 대한 정보를 알 수 있다.

**Comment by sihyung92** [comment]
# 느낀점 & 궁금한 점
- 랜덤엑세스 개념은 DB에서도 중요하게 다뤄지는데, 파일시스템에 대해 깊게 파헤치니 흥미로웠다.
- ㅠㅠ 다 못해서 아쉽다..

# 문제
- FAT File System에서 디스크 직접 접근이 가능한 이유는 무엇인가?

**Reply by 1chz** [reply]
FAT(File Allocation Table) 전체를 메모리에 올려두었으므로

**Reply by inseo24** [reply]
fat 테이블 전체를 메모리에 올림
